infrastructures/docker/infrastructure/src/main/java/org/eclipse/che/workspace/infrastructure/docker/DockerInternalRuntime.java

```diff
@@ -322,6 +354,37 @@ public void accept(String serverRef) {
     }
   }
 
+  private class ServerLivenessHandler implements Consumer<ProbeResult> {
+    @Override
+    public void accept(ProbeResult probeResult) {
+      String machineName = probeResult.getMachineName();
+      DockerMachine machine = startSynchronizer.getMachines().get(machineName);
+      if (machine == null) {
+        // Probably machine was removed from the list during server check start due to some reason
+        return;
+      }
+      String serverName = probeResult.getServerName();
+      ProbeStatus probeStatus = probeResult.getStatus();
+      Server server = machine.getServers().get(serverName);
+      ServerStatus oldServerStatus = server.getStatus();
+      ServerStatus serverStatus;
+
+      if (probeStatus == ProbeStatus.FAILED && oldServerStatus == ServerStatus.RUNNING) {
+        serverStatus = ServerStatus.STOPPED;
+      } else if (probeStatus == ProbeStatus.PASSED
+          && (oldServerStatus == null
```

Comment:
Consider simplifying this check to  ` oldServerStatus != ServerStatus.RUNNING`.

Reply:
tnx